state_machine/docs/README.md

```diff
@@ -0,0 +1,66 @@
+# State Machine Module
+
+This document outlines the purpose of this module, its components and how they all interact with the other modules.
+
+## Contents <!-- omit in toc -->
+- [State Machine Module](#state-machine-module)
+  - [Overview](#overview)
+  - [Code Structure](#code-structure)
+  - [High Level Architecture](#high-level-architecture)
+    - [FSM primer](#fsm-primer)
+  - [Current State Machine Definition](#current-state-machine-definition)
+
+## Overview
+
+The `StateMachine` module implements a FSM (Finite State Machine) that is responsible for managing the node lifecycle since its internal behaviour can be different depending on certain conditions that are used to determine the current state.
```

Comment:
We have an FSM (Finite State Machine) to run a RSM (Replicated State Machine): https://en.wikipedia.org/wiki/State_machine_replication

Reply:
Correct

![image](https://user-images.githubusercontent.com/29378614/218770289-68382d2c-7a2e-4a50-bbd2-32b0a0219a3b.png)